cvat/apps/engine/views.py

```diff
@@ -632,18 +633,22 @@ def _get_rq_response(queue, job_id):
     @action(detail=True, methods=['GET'], serializer_class=DataMetaSerializer,
         url_path='data/meta')
     def data_info(request, pk):
-        db_task = models.Task.objects.prefetch_related('data__images__related_files').select_related('data__video').get(pk=pk)
+        db_task = models.Task.objects.prefetch_related(
+            Prefetch('data', queryset=models.Data.objects.prefetch_related(
+                Prefetch('images', queryset=models.Image.objects.prefetch_related('related_files').order_by('frame'))
+            ))
+        ).get(pk=pk)
 
         if hasattr(db_task.data, 'video'):
             media = [db_task.data.video]
         else:
-            media = list(db_task.data.images.order_by('frame'))
+            media = list(db_task.data.images.all())
```

Comment:
Do not we need to order it? It looks like frame_meta list can have the wrong order in this case

Reply:
> Do not we need to order it? It looks like frame_meta list can have the wrong order in this case

Images are already ordered in the prefetch query
```python
In [12]: db_task = Task.objects.prefetch_related(Prefetch('data', queryset=Data.objects.prefetch_related(Prefetch('images', queryset=Image.objects.prefetch_related('related_files').order_by('frame'))))).get(pk=11)

In [13]: db_task.data.images.all().ordered
Out[13]: True
```

Explicit `order_by` leads to new database queries because methods such as` order_by`, `filter`, etc invalidate prefetched results